csrc/device_lower/pass/replace_size.cpp

```diff
@@ -52,18 +52,40 @@ std::unordered_map<Val*, Val*> getSimplificationMap(Fusion* fusion) {
 
   std::unordered_map<Val*, Val*> simplification_map;
 
+  auto get_number_of_defs = [](Val* val) -> int64_t {
```

Comment:
This might be useful in other contexts. Maybe we could make it a utility like `ir_utils::getOperationCount(Val*)`